From c066a982d1aa1cc21fce45b71f363d9baf19aed6 Mon Sep 17 00:00:00 2001
From: Andrew kim <andrew.kim@intel.com>
Date: Thu, 14 Feb 2019 12:05:14 -0800
Subject: [PATCH 2289/2328] Revert "Fix kernel_read call param"

This reverts commit a1d499debe6cb0dc8bf2dbb14cfb7ccdb3a1e7a9.
---
 drivers/net/wireless/bcmdhd/dhd_linux.c | 8 +++-----
 drivers/net/wireless/bcmdhd/dhd_pcie.c  | 3 +--
 drivers/net/wireless/bcmdhd/linux_osl.c | 2 +-
 3 files changed, 5 insertions(+), 8 deletions(-)

diff --git a/drivers/net/wireless/bcmdhd/dhd_linux.c b/drivers/net/wireless/bcmdhd/dhd_linux.c
index 2786a66..0353fa3 100644
--- a/drivers/net/wireless/bcmdhd/dhd_linux.c
+++ b/drivers/net/wireless/bcmdhd/dhd_linux.c
@@ -10146,7 +10146,7 @@ dhd_os_get_image_block(char *buf, int len, void *image)
 		return 0;
 
 	size = i_size_read(file_inode(fp));
-	rdlen = kernel_read(fp, buf, MIN(len, size), &fp->f_pos);
+	rdlen = kernel_read(fp, fp->f_pos, buf, MIN(len, size));
 
 	if (len >= size && size != rdlen) {
 		return -EIO;
@@ -12807,7 +12807,6 @@ void dhd_get_memdump_info(dhd_pub_t *dhd)
 	uint32 mem_val = DUMP_MEMFILE_MAX;
 	int ret = 0;
 	char *filepath = MEMDUMPINFO;
-	loff_t pos=0;
 
 	/* Read memdump info from the file */
 	fp = filp_open(filepath, O_RDONLY, 0);
@@ -12815,7 +12814,7 @@ void dhd_get_memdump_info(dhd_pub_t *dhd)
 		DHD_ERROR(("%s: File [%s] doesn't exist\n", __FUNCTION__, filepath));
 		goto done;
 	} else {
-		ret = kernel_read(fp, (char *)&mem_val, 4, &pos);
+		ret = kernel_read(fp, 0, (char *)&mem_val, 4);
 		if (ret < 0) {
 			DHD_ERROR(("%s: File read error, ret=%d\n", __FUNCTION__, ret));
 			filp_close(fp, NULL);
@@ -13010,7 +13009,6 @@ void dhd_get_assert_info(dhd_pub_t *dhd)
 {
 	struct file *fp = NULL;
 	char *filepath = ASSERTINFO;
-	loff_t pos=0;
 
 	/*
 	 * Read assert info from the file
@@ -13024,7 +13022,7 @@ void dhd_get_assert_info(dhd_pub_t *dhd)
 		DHD_ERROR(("%s: File [%s] doesn't exist\n", __FUNCTION__, filepath));
 	} else {
 		int mem_val = 0;
-		int ret = kernel_read(fp, (char *)&mem_val, 4, &pos);
+		int ret = kernel_read(fp, 0, (char *)&mem_val, 4);
 		if (ret < 0) {
 			DHD_ERROR(("%s: File read error, ret=%d\n", __FUNCTION__, ret));
 		} else {
diff --git a/drivers/net/wireless/bcmdhd/dhd_pcie.c b/drivers/net/wireless/bcmdhd/dhd_pcie.c
index 7438d58..7577b8e 100644
--- a/drivers/net/wireless/bcmdhd/dhd_pcie.c
+++ b/drivers/net/wireless/bcmdhd/dhd_pcie.c
@@ -1068,14 +1068,13 @@ dhd_get_memcheck_info(void)
 	uint32 mem_val = 0;
 	int ret = 0;
 	char *filepath = MEMCHECKINFO;
-	loff_t pos=0;
 
 	fp = filp_open(filepath, O_RDONLY, 0);
 	if (IS_ERR(fp)) {
 		DHD_ERROR(("[WIFI_SEC] %s: File [%s] doesn't exist\n", __FUNCTION__, filepath));
 		goto done;
 	} else {
-		ret = kernel_read(fp, (char *)&mem_val, 4, &pos);
+		ret = kernel_read(fp, 0, (char *)&mem_val, 4);
 		if (ret < 0) {
 			DHD_ERROR(("[WIFI_SEC] %s: File read error, ret=%d\n", __FUNCTION__, ret));
 			filp_close(fp, NULL);
diff --git a/drivers/net/wireless/bcmdhd/linux_osl.c b/drivers/net/wireless/bcmdhd/linux_osl.c
index f2abdb4..74b65777 100644
--- a/drivers/net/wireless/bcmdhd/linux_osl.c
+++ b/drivers/net/wireless/bcmdhd/linux_osl.c
@@ -2046,7 +2046,7 @@ osl_os_get_image_block(char *buf, int len, void *image)
 	if (!image)
 		return 0;
 
-	rdlen = kernel_read(fp, buf, len, &fp->f_pos);
+	rdlen = kernel_read(fp, fp->f_pos, buf, len);
 	if (rdlen > 0)
 		fp->f_pos += rdlen;
 
-- 
2.7.4

